### PR TITLE
Added Unit tests in build spec

### PR DESCRIPTION
**Pull request by arnab-p**

### CODE REVIEW

**Review by SumanthPuram — COMMENTED** [review]
Let's remove unsafe-perm check and also remove the postInstall script?

File: buildspec.saas.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ phases:
     commands:
       - VERSION="$(date '+%d%m%Y.%H%M%S')"
       - ls
+      - npm install --unsafe-perm
+      - npm test
```
Comment (snarkychef) [inline buildspec.saas.yml:18]: We are running tests on AWS build container and building the application in Dockerfile on alpine. There can be disparity between the builds. Maybe we should run tests in Dockerfile itself?

@SumanthPuram @sivashanmukh What do you guys think?

Comment (sivashanmukh) [reply buildspec.saas.yml:18]: Yea. Makes sense. We should move it to DockerFile

File: Dockerfile [hunk 2]
```diff
@@ -7,9 +7,9 @@ RUN mkdir -p /home/node/app/node_modules && chown -R node:node /home/node/app
 WORKDIR /home/node/app
 USER node
 
-COPY package*.json ./
-COPY build.js ./
-RUN npm install --unsafe-perm
+ADD . /home/node/app
```
Comment (sayan-mitra) [inline Dockerfile:10]: should we run the tests on the image after it's built with docker run?

File: buildspec.master.yml [hunk 3]
```diff
@@ -11,8 +11,7 @@ phases:
     commands:
       - VERSION=master
       - ls
-      - npm install --unsafe-perm
-      - npm test
+      - npm install
```
Comment (sayan-mitra) [inline buildspec.master.yml:14]: is this step required if we do npm install in dockerfile?

File: package.json [hunk 4]
```diff
@@ -37,7 +37,7 @@
     "handlebars": "^4.7.6",
     "is": "^3.1.0",
     "is-ip": "^3.1.0",
-    "isolated-vm": "^3.3.3",
+    "isolated-vm": "^2.0.1",
```
Comment (sayan-mitra) [inline package.json:40]: should we fix the isolate-vm version?
also if we are not committing package-lock file, should we remove it from the source?

File: Dockerfile (hunk 2 above)
Comment (snarkychef) [reply Dockerfile:10]: Sure. Let's do this. Great suggestion

File: buildspec.saas.yml (hunk 1 above)
Comment (SumanthPuram) [reply buildspec.saas.yml]: Definitely a great option if it's not a big effort.

File: package.json (hunk 4 above)
Comment (SumanthPuram) [reply package.json:40]: We should commit package-lock.json file unless we have a good reason not to. 
package-lock should automatically take care of this.

**Comment by sayan-mitra** [comment]
looks good 👍

**Review by sayan-mitra — APPROVED** [review]

**Pull request merged**